documents/system-design-document.md

```diff
@@ -0,0 +1,455 @@
+# System Design Document for TODO List Web App
+
+## 1. Overview
+  The objective of this project is to develop a user-friendly TODO list web application that empowers users to efficiently manage their tasks. The application will provide core functionalities such as creating, reading, updating, deleting, searching, and filtering tasks within a secure environment.
+
+  For detailed project requirements, please refer to the [Product Requirements Document (PRD)](https://github.com/what-a-cool-team/docs/blob/main/prd/what-a-todo.md).
+
+## 2. Functional Requirements
+| Priority | Requirements |
+|----------|--------------|
+| P0 | Users are able to create an account |
+| P0 | Users are able to delete their account |
+| P0 | Users are able to log in securely |
+| P0 | Users are able create new tasks with a name, and optionally add description and tags |
+| P0 | Users are able to update existing tasks to change details such as name, description, status, and tags|
+| P1 | Users are able to view a list of all their tasks |
+| P2 | Users are able to delete tasks they no longer need |
+| P2 | Users are able to search tasks by name or tags |
+| P2 | Users are able to filter tasks by status |
+
+## 3. Operational requirements
+  * **Security**: 
+    * The system should grant users access only to the tasks that they own.
+    * The system should allow users to log in and out using their registered ID and password.
+    * The system should allow users to delete their accounts.
```

Comment:
This seems to be the same thing as the first two of the functional requirements.